### PR TITLE
Adapter Changes for Vungle SDK 7.0.0

### PR DESCRIPTION
**Pull request by stanley-vungle**
This PR is integrating Vungle (Liftoff) 7.0 SDK with the Admob adapter.

### CODE REVIEW

File: adapters/Vungle/VungleAdapter/VungleRouterConsent.m
Comment (joshuajplagonera) [inline adapters/Vungle/VungleAdapter/VungleRouterConsent.m:33]: If these methods are static and can be called at any time (before or after initialization), I think publishers can just call them directly instead of needing a wrapper API?
If so, then we can just remove this class entirely.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m
Comment (joshuajplagonera) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:83]: We can probably remove this method and just call `[VungleAds isInitialized]` directly.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m
Comment (joshuajplagonera) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:87]: Similar to this, I think we can just directly call `[VungleAds getBiddingToken]` instead of needing this util method.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.h
Comment (joshuajplagonera) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.h:37]: Instead of creating a separate utility method for every adapter error code, we can just create one utility method that takes in an adapter error code.

See https://github.com/googleads/googleads-mobile-ios-mediation/blob/main/adapters/Mintegral/MintegralAdapter/GADMAdapterMintegralUtils.m#L20 for reference.

**Review by joshuajplagonera — CHANGES_REQUESTED** [review]
Initial review.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:179]: The errors from Vungle SDK has been updated to have more precise error codes. So it will have a lot more values than the ones listed on the doc and newer codes may be made in the future.

So if this is removed, then the [documentation here](https://developers.google.com/admob/ios/mediation/vungle#error_codes) would need to be removed or updated to the latest. I think having the adapter map it to an existing code would be simpler on both our ends.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.h
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.h:37]: There is a single utility method that takes in an adapter error code.
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.m#L18

Those utility method are convenience methods because the code and message is the same and it is utilized at multiple locations. These convenience methods use the one utility method. For example.
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleInterstitial.m#L95
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m#L62

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.h
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.h]: Updated.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m]: Updated.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m]: Updated.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m]: Updated.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:173]: Updated.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:153]: Yes. It can.
The Vungle SDK doesn't have an "invalid" in the banner size enum so it would default to our regular size banner and then would fail presentation because of a size mismatch instead of failing the load request.

Before I change it, can I get a thought about whether it is better to fail on the load vs the playback?

File: adapters/Vungle/VungleAdapter/VungleRouterConsent.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/VungleRouterConsent.m:33]: The methods in the Vungle SDK are static. We have a publisher request to be able to set the privacy values through the adapter code rather than communicating directly with the SDK API, which was why this class was created before to set the GDPR status. The RouterConfiguration class was removed to condense the IDFV setting to this class too.

File: adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h
Comment (joshh-devrel) [inline adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h:47]: Since all of these properties have been marked deprecated, can you confirm if they aren't supported anymore? Are there substitutions?

(If they aren't supported anymore, this section in the dev doc won't be valid anymore: https://developers.google.com/admob/ios/mediation/vungle#network-specific_parameters)

File: adapters/Vungle/Public/Headers/VungleRouterConsent.h
Comment (joshh-devrel) [inline adapters/Vungle/Public/Headers/VungleRouterConsent.h:27]: When should these consent values be passed to the Vungle SDK? Is it before Vungle SDK initializes or at ad request time?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:185]: Will this callback be called when banner ad is presented? Or when the modal is presented after the banner ad is clicked?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:182]: Is this no-op or just there isn't an equivalent callback in Google Mobile Ads SDK?

If the case is latter, we should update it to:
```
    // Google Mobile Ads SDK doesn't have a matching event.
```

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleInterstitial.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleInterstitial.m:125]: Question, do this callback and `interstitialAdDidTrackImpression` call at the same time?

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m:32]: Nit:
```
/// Vungle banner ad instance.
```

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m:35]: Maybe can update to below:
```
/// UIView that contains Vungle banner ad.
```

File: adapters/Vungle/VungleAdapter/VungleRouterConsent.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/VungleRouterConsent.m:29]: What if the publisher hasn't specified COPPA value?
https://developers.google.com/admob/ios/targeting#child-directed_setting

Should we just not set the COPPA value in the Vungle SDK?

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m:65]: Why do we have the below for bidding but not waterfall? Can we just include `vungleAdSizeForAdSize` in utils and use it for both bidding and waterfall?
`_bannerSize = [self vungleAdSizeForAdSize:adConfiguration.adSize];`

Also the method name with `filter` sounds odd to me.

File: adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h
Comment (stanley-vungle) [reply adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h:47]: Yes. Those properties are no longer supported anymore without any substitutions.

File: adapters/Vungle/Public/Headers/VungleRouterConsent.h
Comment (stanley-vungle) [reply adapters/Vungle/Public/Headers/VungleRouterConsent.h:27]: It can be anytime. But for it to take effect, it needs to happen before the ad request is made.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m:185]: This callback will be called when the banner ad is presented. We do not have an equivalent callback for when the modal is presented after the banner ad is clicked.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m]: Done.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleInterstitial.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleInterstitial.m:125]: No. It should be very close. The DidPresent callback is called when the view is successfully attached, and the impression callback is called when our ad plays successfully and sends a signal to the Vungle SDK to trigger.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m]: Done.

File: adapters/Vungle/VungleAdapter/VungleRouterConsent.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/VungleRouterConsent.m:29]: I will remove this wrapper around the Vungle COPPA. There was a previous comment somewhere that the publisher should use the Google SDK API to set the COPPA flag and the adapter would attempt to pull the childDirectedTreatment flag from the Google SDK on every load request.
https://github.com/Vungle/googleads-mobile-ios-mediation/blob/7.0.x_pre_release/adapters/Vungle/VungleAdapter/GADMediationAdapterVungle.m#L111

But I just moved the method away from the public header at the time.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleBanner.m]: Ok. Done.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m]: Moved this method to utils to follow review from joshh.

**Comment by joshh-devrel** [comment]
@stanley-vungle Can you rebase this pull request as we just finished the rebrand to Liftoff Monetize?

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m
Comment (joshuajplagonera) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m:29]: Is there a reason this class is still here? I think this class can be removed.
It looks like the publishers can directly call these `VunglePrivacySettings` methods statically and we can update the docs to clarify they need to be set before ad request time.

Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m:29]: @joshuajplagonera Another reason why it was wanted was to avoid less direct interaction with our SDK so if the API changes, the migration for the publisher would be easier.

I got the approval to remove this if this is a blocker for the certification.

Comment (joshuajplagonera) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m:29]: Thanks for the explanation. This can be kept. :)

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m
Comment (joshuajplagonera) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m:120]: We're having build issues where `VungleAdsExtras` cannot be found:

```
Use of undeclared identifier 'VungleAdsExtras'
```

Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m:120]: This was support for the watermark. I thought that it was going to be released soon on our end so I included this back in. Let me roll this commit back.

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m:178]: Can you remind me again when would/should this `initialized` callback be called? Do we really need this callback?

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleUtils.m
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleUtils.m:86]: Can you clarify why is that the case?

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.h
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.h:21]: It's still "Vungle SDK" in [Liftoff's documentation](https://support.vungle.com/hc/en-us/articles/360002925791), but I will defer to Liftoff team on whether it should be called Vungle or Liftoff Monetize SDK here.

Comment (joshh-devrel) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.h:21]: The same applies to the other places.

File: adapters/LiftoffMonetize/Public/Headers/VungleAdNetworkExtras.h
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/Public/Headers/VungleAdNetworkExtras.h:47]: Seems like these attributes aren't used anywhere? Can we remove them?

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.m
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.m:49]: Just want to confirm if we need to do `if (appID.length)`.

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleBanner.m
Comment (joshh-devrel) [inline adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleBanner.m:110]: Should indicate nullability for the input parameter.

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m
Comment (joshh-devrel) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m]: @stanley-vungle Just to confirm - so watermark for Vungle bidding ads isn't supported yet, correct? Would you know when will it be supported?

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/VungleRouterConsent.m:29]: Ok. I'll remove the class.

The consent values should be set prior to the requesting ads of the Vungle SDK (through the GMA SDK). Currently our SDK does not utilize any identifiers for the initialization but it will be passed when the ad request is made.

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m:178]: This `initialized` callback is called if the Vungle SDK wasn't initialized yet. It is triggered as a form of lazy initialization. If the publisher's app initializes the Vungle SDK before the load request, it will never be hit. If the Vungle SDK hasn't been initialized, the load request will first attempt initialization and on the callback, it will trigger this.

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleUtils.m
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleUtils.m:86]: It is the only supported size from Vungle. Our SDK will try to fit the banner in the view, but the true height size of the assets is 50 px

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.h
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.h:21]: @joshh-devrel In the [re-branding commit](https://github.com/googleads/googleads-mobile-ios-mediation/commit/1bee38f642f6b0cbbf9bbdd0ba66899be3d60a51), almost all comments were changed from `Vungle` to `Liftoff Monetize`. I thought that these cases were missed so I changed it. I'll remove the "SDK" part of the comment.

File: adapters/LiftoffMonetize/Public/Headers/VungleAdNetworkExtras.h
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/Public/Headers/VungleAdNetworkExtras.h:47]: Ok :+1:

File: adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.m
Comment (stanley-vungle) [reply adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMediationAdapterVungle.m:49]: Our SDK has a check internally to see if the string is empty and send an error, but I think your appID.length would handle more integration mishaps. I'll change it.

**Pull request merged**
